### PR TITLE
CH-#123 - RSYSLOG COOKBOOK :: Extending Attributes in lieu of hardcoded values

### PR DESCRIPTION
**Pull request by rmcleod8**
### Description

Removes hardcoded values of $Umask, $FileCreateMode and $DirCreateMode. They can now be set using the attribute.
Added attributes for $DirOwner, $DirGroup and $Templates. These can now be set using attributes.

### Issues Resolved

https://github.com/chef-cookbooks/rsyslog/issues/123

### Check List

- [x] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [ ] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

**Comment by dquiles** [comment]
lgtm :+1:

File: templates/default/rsyslog.conf.erb [hunk 1]
```diff
@@ -76,6 +76,14 @@ $ActionFileDefaultTemplate <%= node["rsyslog"]["default_file_template"] %>
 $ActionFileDefaultTemplate RSYSLOG_TraditionalFileFormat
 <% end -%>
 
+#
```
Comment (tas50) [inline templates/default/rsyslog.conf.erb]: This should be within the if statement so we avoid adding a meaningless comment when users don't have templates set

File: metadata.rb [hunk 2]
```diff
@@ -4,7 +4,7 @@
 license           'Apache 2.0'
 description       'Installs and configures rsyslog'
 long_description  IO.read(File.join(File.dirname(__FILE__), 'README.md'))
-version           '5.0.1'
+version           '5.0.2'
```
Comment (tas50) [inline metadata.rb]: This can stay here, but in the future we have tooling that handles the bump

Comment (rmcleod8) [inline metadata.rb]: good to know, sorry about that... habit.

**Comment by Argoday** [comment]
I don't think this is correct, these properties (Owner, Group, CreateMode) are sticky and apply to all entries that are below them (rsyslog conf files are parsed top-down) - so they should be set just above each entry that they are needed, this PR would change the FileOwner etc.. for all log files instead of only for the ones that are needed

**Comment by rmcleod8** [comment]
@Argoday This change is correcting previously hardcoded and missing attributes which would atleast let the user to set them.

**Comment by Argoday** [comment]
Sry, to be more specific - this change improves the cookbook to allow the top-of-file defaults to be changed, however in regards to implementing {DRQS 93976638} and/or any other request that would want to use these settings - this comes up short.

As-is this enables all log entries to have a single owner however we should we able to set the owner individually for each log entry

**Comment by Argoday** [comment]
@rmcleod8 - retracting my previous comment .. didn't pay enough attention to which repo this is

**Comment by tas50** [comment]
Thanks for the fixes

**Pull request merged**